newIDE/app/src/EventsSheet/ParameterFields/GenericExpressionField/index.js

```diff
@@ -268,12 +268,46 @@ export default class ExpressionField extends React.Component<Props, State> {
     parameterValues: ParameterValues
   ) => {
     if (!this._inputElement) return;
+    const {
+      globalObjectsContainer,
+      objectsContainer,
+      expressionType,
+      value,
+    } = this.props;
     const cursorPosition = this._inputElement.selectionStart;
+    const parser = new gd.ExpressionParser2();
 
-    const functionCall = formatExpressionCall(expressionInfo, parameterValues);
+    // We add a character so that getCompletionDescriptionsFor will always return the type of the node
+    // like it would do for an automatic completion.
+    const expressionNode = parser
+      .parseExpression(
+        value.substring(0, cursorPosition) +
+          'a' +
+          value.substring(cursorPosition)
+      )
+      .get();
+    const completionDescriptions = gd.ExpressionCompletionFinder.getCompletionDescriptionsFor(
+      gd.JsPlatform.get(),
+      globalObjectsContainer,
+      objectsContainer,
+      expressionType,
+      expressionNode,
+      cursorPosition
+    );
+
+    let shouldConvertToString = false;
+    if (expressionInfo.metadata.getReturnType() === 'number') {
```

Comment:
maybe add a comment on this whole part explaining the logic happening:
"If the field expects a string and one of the completions returns a number, then ... so that we can add the ToString()..."